### ISSUE TITLE
Potential bug in MappedCart (in currently unused code)

### ISSUE THREAD

**Issue by manauref** [ammarhakim/gkyl#36]
I think there's an erroneous memory access in line 71 of Grid/MappedCart.lua.

It is assigning a value to d1[3] and d2[3], but these vectors have 2 dimensions, not three. I suspect that the correct code that should replace lines 69-71 is:
```
   d1[1], d2[1], d3[1] = diff.derivt(self._mapc2p, 1)(xc)
   d1[2], d2[2], d3[2] = diff.derivt(self._mapc2p, 2)(xc)
```

Otherwise the current code takes the derivative with respect to a third variable that doesn't exist. If this is correct, I wonder why test_2() in Unit/test_MappedCart.lua works. Maybe I'm wrong and this issue should be resolved/closed.

**Comment by ammarhakim** [comment]
Yes this looks wrong. First thing to do is add a regression/Unit test for this mapping. The current test is not actually checking the metric tensor or Jacobian and just checks that the object was created. I will extend the test and we can then fix.
